compiler/noirc_frontend/src/elaborator/expressions.rs

```diff
@@ -430,6 +430,24 @@ impl<'context> Elaborator<'context> {
             }
         };
 
+        let struct_generics = if let Some(turbofish_generics) = &last_segment.generics {
```

Comment:
Below this we still clone the `struct_generics` on line 452 - seems that should be unneeded now, right?

Reply:
I think it's still needed, `resolve_turbofish_generics` takes `Vec<UnresolvedType>`. I tried passing a reference, but that ends up calling `resolve_type_inner` which takes `UnresolveType`. I tried changing that too to a reference but at some point the type and not the rereference is needed... so I gave up.

Reply:
Ah, ok. I thought this new variable we created on line 433 was already a `Vec<UnresolvedType>` rather than just a reference to one.

Reply:
Just pulled the branch myself and found that this is a value and its just that we use both clones of it.
Some things are harder to see on github I suppose.